docs/source/tutorials/adascale.rst

```diff
@@ -0,0 +1,89 @@
+AdaScale SGD
+============
+
+`AdaScale <https://arxiv.org/pdf/2007.05105.pdf>`_ adaptively scales the learning rate when using larger batch sizes for data-parallel training. Let's suppose that your trainer looks like
+
+.. code-block:: python
+
+
+    import torch
+    from torch.nn.parallel import DistributedDataParallel as DDP
+
+
+    def train(
+        rank: int,
+        world_size: int,
+        epochs: int):
+
+        # DDP
+        dist_init(rank, world_size)
+
+        # Problem statement
+        model = myAwesomeModel().to(rank)
+        model = DDP(model, device_ids=[rank])
+        dataloader = mySuperFastDataloader()
+        loss_ln = myVeryRelevantLoss()
+
+        # optimizer specific arguments e.g. LR, momentum, etc...
+        base_optimizer_arguments = { "lr": 1e-4}
+        optimizer = torch.optim.SGD(
+            params=model.parameters(),
+            **base_optimizer_arguments)
+
+        # Any relevant training loop. For example:
+        model.train()
+        for e in range(epochs):
+            for (data, target) in dataloader:
+                data, target = data.to(rank), target.to(rank)
+                # Train
+                model.zero_grad()
```

Comment:
Hi @aurickq, is there any specific reason we use model.zero_grad not optimizer.zero_grad as I see in most other code?

Reply:
I copied the same example snippet from the OSS tutorial, which also used `model.zero_grad`. I guess another reason is the `AdaScale` wrapped optimizer doesn't have a `zero_grad` method, but there should be easy solutions for that (either set `patch_optimizer=True` and use the original optimizer, or assign the wrapped optimizer to a different variable).

Reply:
Thanks for the explanation! BTW, feel free to review PRs on AdaScale in this repo if you want!